### PR TITLE
Improve captions appearance when there is also a CTA

### LINKED ISSUE #13425: CTA can sometimes cover captions
## Bug Description

Reported at: https://wordpress.org/support/topic/cta-covers-subtitles/

## Expected Behaviour

No overlap

## Screenshots

PHOTO-2023-06-22-21-39-07

## Additional Context

- Plugin Version: 1.33.0
- WordPress Version: 6.2.2
- Affected device: Pixel 2, iPhone X, also in the Web Stories development mode "emulator"

### PR DESCRIPTION
**Pull request by AnuragVasanwala**
The goal of the PR is to fix layout of the video subtitles (when present) so that they won't display below CTA button.

## Summary

`amp-story-captions` adds caption in shadow-dom. Thus, parent `div` container of the `amp-story-captions` won't reflect the actual size of the `amp-story-captions`.

This will fix this issue by setting `max-height` of the `amp-story-grid-layer.align-bottom` exactly to the shadow-dom component.

Additionally, fixed-height layout of the `amp-story-captions` is changed to container in order to occupy only required space.

## User-facing changes

Not applicable.

## Testing Instructions

Follow #13425 for the testing instruction.

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #13425

### CODE REVIEW

**Comment by swissspidy** [comment]
Any thoughts on trying out `style-preset="default"` as suggested on the ticket?

**Comment by AnuragVasanwala** [comment]
> Any thoughts on trying out `style-preset="default"` as suggested on the ticket?

This won't fix overlay issue because `amp-story-page-outlink` adds CTA button inside shadow-dom. This creates problem because height of the shadow-dom element (here, CTA button) remains `ZERO` and thus `amp-story-captions` for some devices get's below CTA button:
image

I will provide in depth analysis and fix on the PR by adding detailed video GIFs representing issue and how it's been fixed.

Correct me if I have missed something.

**Comment by AnuragVasanwala** [comment]
> Any thoughts on trying out `style-preset="default"` as suggested on the ticket?

@swissspidy I tried `style-preset="default"` with all possible styling updates. Unfortunately, captions are not horizontally centered without `captions-area` wrapper when `style-preset="default"` is used.

| | With `captions-area` wrapper | Without `captions-area` wrapper |
|---|:-:|:-:|
| **[1]** without `style-preset = "default"` | **[1.1]** mobile | **[1.2]** mobile (1) |
| **[2]** with `style-preset = "default"` | **[2.1]** mobile (3) | **[2.2]** mobile (2) |

May we go with **[2.1]** approach?

File: packages/output/src/utils/styles.js [hunk 1]
```diff
@@ -135,6 +135,7 @@ function CustomStyles() {
               amp-story-grid-layer.align-bottom {
                 align-content: end;
                 padding: 0;
+                max-height: calc(100vh - 148px);
```
Comment (swissspidy) [inline packages/output/src/utils/styles.js]: Let's add a comment explaining where 148px is coming from.

IIRC it's 2x the size of the CTA layer, so perhaps we can even do `calc(100vh - ( 2 * 74px) );` to make that easier to understand.

**Comment by swissspidy** [comment]
Interesting! Thanks for sharing the various options, much appreciated!

Let's try [2.1] indeed 👍

**Comment by swissspidy** [comment]
Note that two snapshot tests in `packages/output/src/test/page.js` are failing now.

And that one Karma test is just flakey. Maybe we can make it more robust at some point.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**